presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfMetadataWriter.java

```diff
@@ -222,20 +224,22 @@ public static DwrfProto.ColumnStatistics toColumnStatistics(ColumnStatistics col
                     .build());
         }
 
-        if (columnStatistics.getIntegerStatistics() != null) {
-            DwrfProto.IntegerStatistics.Builder integerStatistics = DwrfProto.IntegerStatistics.newBuilder()
-                    .setMinimum(columnStatistics.getIntegerStatistics().getMin())
-                    .setMaximum(columnStatistics.getIntegerStatistics().getMax());
-            if (columnStatistics.getIntegerStatistics().getSum() != null) {
-                integerStatistics.setSum(columnStatistics.getIntegerStatistics().getSum());
+        IntegerStatistics integerStatistics = columnStatistics.getIntegerStatistics();
+        if (integerStatistics != null) {
+            DwrfProto.IntegerStatistics.Builder dwrfIntegerStatistics = DwrfProto.IntegerStatistics.newBuilder()
+                    .setMinimum(integerStatistics.getMinPrimitive())
```

Comment:
setMinimum uses primitive long, so this change is fine.

Reply:
Sorry, this is not visible in the PR UI, but all min/max/sum setters in DwrfProto are using primitive types. Meaning that an attempt to unbox a null value returned by the old version `Long getMin/Max/Sum` would result in NPE. There is a parity in logic because the getMin/Max/SumPrimitive checks that the presence of the value before returning a  primitive value.